core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala

```diff
@@ -108,19 +108,49 @@ class MetadataRequestTest extends BaseRequestTest {
 
     // v0, Doesn't support a "no topics" request
     // v1, Empty list represents "no topics"
-    val metadataResponse = sendMetadataRequest(new MetadataRequest(List[String]().asJava, 1.toShort))
+    val metadataResponse = sendMetadataRequest(new MetadataRequest(List[String]().asJava, true, 1.toShort))
     assertTrue("Response should have no errors", metadataResponse.errors.isEmpty)
     assertTrue("Response should have no topics", metadataResponse.topicMetadata.isEmpty)
   }
 
   @Test
+  def testAutoTopicCreation(): Unit = {
+    def checkAutoCreatedTopic(existingTopic: String, autoCreatedTopic: String, response: MetadataResponse): Unit = {
+      assertNull(response.errors.get(existingTopic))
+      assertEquals(Errors.LEADER_NOT_AVAILABLE, response.errors.get(autoCreatedTopic))
+      assertEquals(Some(servers.head.config.numPartitions), zkUtils.getTopicPartitionCount(autoCreatedTopic))
+      for (i <- 0 until servers.head.config.numPartitions)
+        TestUtils.waitUntilMetadataIsPropagated(servers, autoCreatedTopic, i)
+    }
```

Comment:
Is there a race condition here where we might not get Errors.LEADER_NOT_AVAILABLE, if the brokers are quicker than usual?